DSCResources/MSFT_xDhcpClient/MSFT_xDhcpClient.psm1

```diff
@@ -0,0 +1,173 @@
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData -StringData @'
+GettingDHCPClientMessage=Getting the DHCP Client on {1} interface "{0}".
+ApplyingDHCPClientMessage=Applying the DHCP Client on {1} interface "{0}".
+DHCPClientSetStateMessage=DHCP Client was set to the desired state {2} on {1} interface "{0}".
+CheckingDHCPClientMessage=Checking the DHCP Client on {1} interface "{0}".
+DHCPClientDoesNotMatchMessage=DHCP Client is not in the desired state {2} on {1} interface "{0}".
+InterfaceNotAvailableError=Interface "{0}" is not available. Please select a valid interface and try again.
+'@
+}
+
+function Get-TargetResource
+{
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String] $InterfaceAlias,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('IPv4', 'IPv6')]
+        [String] $AddressFamily,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('Enabled', 'Disabled')]
+        [String] $State
+    )
+
+    Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+        $($LocalizedData.GettingDHCPClientMessage) `
+        -f $InterfaceAlias,$AddressFamily `
+        ) -join '')
+
+    Test-ResourceProperty @PSBoundParameters
+    
+    $CurrentDHCPClient = Get-NetIPInterface `
+        -InterfaceAlias $InterfaceAlias `
+        -AddressFamily $AddressFamily
+
+    $returnValue = @{
+        State          = $CurrentDHCPClient.Dhcp
+        AddressFamily  = $AddressFamily
+        InterfaceAlias = $InterfaceAlias
+    }
+
+    $returnValue
+}
+
+function Set-TargetResource
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String] $InterfaceAlias,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('IPv4', 'IPv6')]
+        [String] $AddressFamily,
+
+        [Parameter(Mandatory)]
+        [ValidateSet('Enabled', 'Disabled')]
+        [String] $State
+    )
+
+    Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+        $($LocalizedData.ApplyingDHCPClientMessage) `
+        -f $InterfaceAlias,$AddressFamily `
+        ) -join '')
+
+    Test-ResourceProperty @PSBoundParameters
+    
+    $CurrentDHCPClient = Get-NetIPInterface `
+        -InterfaceAlias $InterfaceAlias `
+        -AddressFamily $AddressFamily
+        
+    # The DHCP Client is in a different state - so change it.
+    if ($CurrentDHCPClient.DHCP -ne $State)
```

Comment:
It really is redundant TBH - but I put in in there "just-in-case" there was a flaw in my logic somewhere. Not that it would actually cause a problem should this set fire when it didn't need to. I'll remove it :)